onnxruntime/test/providers/qnn/qnn_basic_test.cc

```diff
@@ -1100,10 +1119,60 @@ TEST_F(QnnHTPBackendTests, CastAddHTPAccuracyTest) {
 #endif
```

Comment:
should be replace by
provider_options["backend_type"] = "htp";
seems we didn't clean up all the  places like this.